code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -295,6 +342,39 @@
 		busy = SPIDER_IDLE
 		stop_automated_movement = FALSE
 
+/mob/living/simple_animal/hostile/poison/giant_spider/Login()
+	..()
+	GLOB.spidermobs += src
+
+/mob/living/simple_animal/hostile/poison/giant_spider/Logout(GLOB.spidermobs)
+	..()
+	GLOB.spidermobs -= src
+
+/datum/action/innate/spider/comm
+	name = "Command"
+	button_icon_state = "cult_comms"
+
+/datum/action/innate/spider/comm/IsAvailable()
+	if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife))
+		return FALSE
+	return TRUE
+
+/datum/action/innate/spider/comm/Trigger()
+	var/input = stripped_input(usr, "Input a message for your legions to follow.", "Command", "")
+	if(QDELETED(src) || !input || !IsAvailable())
+		return FALSE
+	spider_command(usr, input)
+	return TRUE
+
+/datum/action/innate/spider/comm/proc/spider_command(mob/living/user, message)
+	var/my_message

```

Comment:
why is this declared up here?

Reply:
he had some thing for deadchat last night where it was needed but I removed it without removing the var, whoops :p

Reply:
Yeah, so move it down to there

Reply:
you are too smart for your own good cyberman!